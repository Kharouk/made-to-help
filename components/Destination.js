import React from 'react';
import { View, Text } from 'react-native';
import { GooglePlacesAutocomplete } from 'react-native-google-places-autocomplete';
import KEY from '../env.config';

export default class Destination extends React.Component {
  constructor(props) {
    super(props);
    this.state = { latitude: '', longitude: '' };
    this.savingLocation = this.savingLocation.bind(this);
  }

  savingLocation(latitude, longitude) {
    this.setState(
      {
        latitude,
        longitude,
      },
    );
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <Text style={{ flex: 0.10 }}>
          {`${this.state.longitude} - ${this.state.latitude}`}
        </Text>
        <GooglePlacesAutocomplete
          placeholder="Destination"
          minLength={2}
          autoFocus={false}
          returnKeyType="search"
          listViewDisplayed="auto"
          fetchDetails
          onPress={(data, details = null) => {
            this.savingLocation(details.geometry.location.lat, details.geometry.location.lng);
          }}

          getDefaultValue={() => ''}

          query={{
            key: KEY,
            language: 'en',
          }}

          styles={{
            textInputContainer: {
              width: '100%',
            },
            description: {
              fontWeight: 'bold',
            },
            predefinedPlacesDescription: {
              color: '#1faadb',
            },
            listView: {
              color: 'black',
              zIndex: 16,
            },
          }}
          nearbyPlacesAPI="GooglePlacesSearch"
          GoogleReverseGeocodingQuery={{
          }}
          GooglePlacesSearchQuery={{
            rankby: 'distance',
          }}

          filterReverseGeocodingByTypes={['locality', 'administrative_area_level_3']}
          debounce={200}
          currentLocation
        />
      </View>

    );
  }
}

// AppRegistry.registerComponent('made-to-help', () => Directions);
