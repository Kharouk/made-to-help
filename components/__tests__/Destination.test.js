import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Destination from '../Destination';

Enzyme.configure({ adapter: new Adapter() });

test('Renders Destination component', () => {
  const snapshot = renderer.create(<Destination />).toJSON()
  expect(snapshot).toMatchSnapshot();
})

it('Should test for setState changes', () => {
  const wrapper = shallow(<Destination />);
  const instance = wrapper.instance()
  instance.savingLocation(-0.09, 51)
  expect(instance.state.latitude).toBe(-0.09)
  expect(instance.state.longitude).toBe(51)
})
