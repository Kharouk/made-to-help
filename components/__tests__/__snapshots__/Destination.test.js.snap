// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Destination component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "flex": 0.1,
      }
    }
  >
     - 
  </Text>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#C9C9CE",
            "borderBottomColor": "#b5b5b5",
            "borderBottomWidth": 0.5,
            "borderTopColor": "#7e7e7e",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "height": 44,
          },
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoFocus={false}
        clearButtonMode="while-editing"
        editable={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Destination"
        placeholderTextColor="#A8A8A8"
        returnKeyType="search"
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "flex": 1,
              "fontSize": 15,
              "height": 28,
              "marginLeft": 8,
              "marginRight": 8,
              "marginTop": 7.5,
              "paddingBottom": 4.5,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 4.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;
